src/test_helpers/xds.rs

```
@@ -0,0 +1,179 @@
+        watch::Sender<Result<DiscoveryResponse, tonic::Status>>,
+        watch::Sender<Result<DeltaDiscoveryResponse, tonic::Status>>,
+        AdsClient,
+    ) {
+        let delta_default = Err(tonic::Status::not_found("mock not set"));
+        let (delta_tx, delta_rx) = watch::channel(delta_default);
+
+        let default = Err(tonic::Status::not_found("mock not set"));
+        let (tx, rx) = watch::channel(default);
+
+        let server = AdsServer { response: rx, delta_response: delta_rx };
+        let listener = tokio::net::TcpListener::bind("127.0.0.1:15012").await.unwrap();
+        let certs = tls::generate_test_certs(
+            &Identity::default(),
+            Duration::from_secs(0),
+            Duration::from_secs(100),
+        );
+        let acceptor = tls::ControlPlaneCertProvider(certs);
+        let tls_stream = crate::hyper_util::tls_server(acceptor, listener);
+        let incoming = hyper::server::accept::from_stream(tls_stream);
+
+        let srv = AggregatedDiscoveryServiceServer::new(server);
+        tokio::spawn(async move {
+            hyper::Server::builder(incoming)
+                .serve(make_service_fn(move |_| {
+                    let mut srv = srv.clone();
+                    future::ok::<_, Infallible>(tower::service_fn(
+                        move |req: hyper::Request<hyper::Body>| srv.call(req),
+                    ))
+                }))
+                .await
+                .unwrap()
+        });
+        let xds_client = cfg.build(metrics, block_ready);
+        (tx, delta_tx, xds_client)
+    }
+}
+
+
+#[async_trait]
+impl AggregatedDiscoveryService for AdsServer {
+    type DeltaAggregatedResourcesStream = Pin<Box<dyn Stream<Item = Result<DeltaDiscoveryResponse, Status>> + Send>>;
+    type StreamAggregatedResourcesStream = Pin<Box<dyn Stream<Item = Result<DiscoveryResponse, Status>> + Send>>;
+    async fn stream_aggregated_resources(
```

we don't use this in ztunnel, so we can just return an error or panic or something